src/utils/RouteAvailabilityHelper.js

```diff
@@ -0,0 +1,20 @@
+import { useContext } from 'react';
+import { AuthContext } from '../context/AuthProvider';
+
+export default function isRouteAvailable(path) {
+  const { userRole } = useContext(AuthContext);
+
+  if ((path === '/jobs' || path === '/job/:id') && userRole === 'user') {
```

Comment:
It is enough if routeAvailability is computed based on the users roles, this feels complicated with no reason.

Reply:
I tried to make it more readable using the mapping strategy as provided above

"This could be written without ifs.
const userRoleMapping = { 'sys_admin': 1, 'company_user': 2, 'user': 3};
and calling userRoleMapping['sys_admin'] would return 1 etc."